web_src/js/markup/anchors.js

```diff
@@ -31,5 +33,18 @@ export function initMarkupAnchors() {
     heading.prepend(a);
   }
 
+  // handle user-defined `name` anchors like `[Link](#link)` linking to `<a name="link"></a>Link`
+  for (const a of document.querySelectorAll('.markup a[href^="#"]')) {
+    const href = a.getAttribute('href');
+    if (!href.startsWith('#user-content-')) continue;
+    const originalId = href.replace(/^#user-content-/, '');
+    a.setAttribute('href', `#${encodeURIComponent(originalId)}`);
+    if (document.getElementsByName(originalId).length !== 1) {
```

Comment:
Wait, `getElementsByName`?
Isn't that the wrong call?
Shouldn't it be `getElementsByID`?

Reply:
As far as I know, we use that method because we are considering the cases that we use `<a name="link"><\a>`. In this condition, we are making it so that those anchors that do not use this notation work as intended by having the event listener.

Reply:
It's correct I'd say. It targets specifically links with `name`. A likely better and more specific selector could be `.markup a[name]` so we don't accidentially select other things.

Reply:
Even better selector: `a.closest('.markup').querySelectorAll('a[name]')`

This will ensure it will only search for the target in the current markup document. I can follow up with this change on master later.

Reply:
Followup: https://github.com/go-gitea/gitea/pull/29679